src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -3821,6 +3820,300 @@ internal new NamedTypeSymbol CreateNativeIntegerTypeSymbol(bool signed)
+                                  BinaryOperatorKind.GreaterThanOrEqual or
+                                  BinaryOperatorKind.LessThanOrEqual &&
+                    csharpReturnType.SpecialType == SpecialType.System_Boolean &&
+                    csharpLeftType is PointerTypeSymbol { PointedAtType.SpecialType: SpecialType.System_Void } &&
+                    csharpRightType is PointerTypeSymbol { PointedAtType.SpecialType: SpecialType.System_Void })
+                {
+                    return;
+                }
+
+                // T* operator+(T*, int/uint/long/ulong i)
+                if (binaryKind is BinaryOperatorKind.Addition &&
+                    csharpLeftType.IsPointerType() &&
+                    isAllowedPointerArithmeticIntegralType(csharpRightType) &&
+                    TypeSymbol.Equals(csharpLeftType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // T* operator+(int/uint/long/ulong i, T*)
+                if (binaryKind is BinaryOperatorKind.Addition &&
+                    csharpRightType.IsPointerType() &&
+                    isAllowedPointerArithmeticIntegralType(csharpLeftType) &&
+                    TypeSymbol.Equals(csharpRightType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // T* operator-(T*, int/uint/long/ulong i)
+                if (binaryKind is BinaryOperatorKind.Subtraction &&
+                    csharpLeftType.IsPointerType() &&
+                    isAllowedPointerArithmeticIntegralType(csharpRightType) &&
+                    TypeSymbol.Equals(csharpLeftType, csharpReturnType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // long operator-(T*, T*)
+                if (binaryKind is BinaryOperatorKind.Subtraction &&
+                    csharpLeftType.IsPointerType() &&
+                    csharpReturnType.SpecialType is SpecialType.System_Int64 &&
+                    TypeSymbol.Equals(csharpLeftType, csharpRightType, TypeCompareKind.ConsiderEverything))
+                {
+                    return;
+                }
+
+                // ROS<byte> operator+(ROS<byte>, ROS<byte>). Legal because of utf8 strings.
+                if (binaryKind is BinaryOperatorKind.Addition &&
+                    isReadOnlySpanOfByteType(csharpReturnType) &&
+                    isReadOnlySpanOfByteType(csharpLeftType) &&
+                    isReadOnlySpanOfByteType(csharpRightType))
+                {
+                    return;
+                }
+
+                throw new ArgumentException(string.Format(CodeAnalysisResources.Unsupported_built_in_operator_0, $"{csharpReturnType.ToDisplayString()} operator {name}({csharpLeftType.ToDisplayString()}, {csharpRightType.ToDisplayString()})"));
```

Comment:
>throw new ArgumentException [](http://example.com/codeflow?start=16&length=27)

Do we want some scenarios involving error types to pass through here? 
We have checks for TypeKind with the goal to identify delegates and enums. If type is an error type (perhaps cannot be found in a new compilation), those checks going to fail. So, we might want to return if at least one operand type (intentionally excluding return type) is an error type with no SpecialType.

Reply:
> Do we want some scenarios involving error types to pass through here?

I'm completely ok with the current system as is.  IDE has no need for the error case as of yet, and none of the compiler test cases hit that.  So i'd rather make any changes here based on a needed use case.  :)

Reply:
> I'm completely ok with the current system as is.  IDE has no need for the error case as of yet, and none of the compiler test cases hit that.

Sounds good. just wanted to make sure a possibility of a failure like that is understood.

Reply:
👍